### ISSUE TITLE
Error with http2-wrapper

### ISSUE THREAD

**Issue by nickluger** [apify/got-scraping#68]
This error was a heisenbug / flaky error in my case, so I cannot guarantee it's reproducible, but I can hint to code causing it.

This was called:
```js
 gotScraping.get({url, proxyUrl, timeout: {request: 30_000}});
```
This error ocurred:
```
RequestError: Cannot read properties of undefined (reading 'end')\n    at Request._beforeError (/app/node_modules/got-cjs/dist/source/core/index.js:333:21)\n    at Request.flush (/app/node_modules/got-cjs/dist/source/core/index.js:322:18)\n    at /app/node_modules/got-scraping/dist/resolve-protocol.js:33:21\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)"
```
**This might be the reason:**
`got-scraping` is calling the `auto` function of `http2-wrapper` [here](https://github.com/apify/got-scraping/blob/6295bc0d1f39fdd4ebccb63a2d4cd1ea534694b0/src/resolve-protocol.ts#L30). 

Since `http2-wrapper` version `2.1.5` the `auto` function is using `delayAsyncDestroy` (see [here](https://github.com/szmarczak/http2-wrapper/blob/6562e77ef7c40d35039ed2fe25a106233375054c/source/auto.js#L195)), which in turn can return `undefined` (see [here](https://github.com/szmarczak/http2-wrapper/blob/6562e77ef7c40d35039ed2fe25a106233375054c/source/utils/delay-async-destroy.js#L5)).

`got-scraping` does not handle this case and the subsequent call to `end()` [here](https://github.com/apify/got-scraping/blob/6295bc0d1f39fdd4ebccb63a2d4cd1ea534694b0/src/resolve-protocol.ts#L37) fails with the described error.

**Workaround**
Pinning the `http2-wrapper` dependency to version `2.1.4` (using yarn resolutions in my case), fixed the issue.

**Reproduction**
For that reason, pinning the  `http2-wrapper` to the latest version (or any version since including `2.1.5`), is necessary to reproduce this, as the caret is used here (` "http2-wrapper": "^2.1.4"`), which can lead to minor+patch later versions.

**Comment by szmarczak** [comment]
Thankfully I'm the author of `http2-wrapper` so I'll fix it in a moment, thanks for reporting!

**Comment by szmarczak** [comment]
Released `http2-wrapper@2.1.11`, should be working now.
